### ISSUE TITLE
[Bug] Issue with merging when line has head anchors

### ISSUE THREAD

**Issue by tuanchauict** [tuanchauict/MonoSketch#194]
```
        ▲     
┌───────┬────┐
│       │    │
│       │    │
│       │    │
│       │    │
└───────┴────┘
        ▼     
```

**Comment by tuanchauict** [comment]
![image](https://user-images.githubusercontent.com/955306/129472617-1d0ce278-1271-4426-b8d8-b411f7e29118.png)
```
[{"type":"R","b":"57|12|37|8","e":{"fu":"F1","be":true,"bu":"B1"}},{"type":"R","b":"62|10|22|6","e":{"fe":true,"fu":"F2","be":true,"bu":"B1"}},{"type":"R","b":"71|7|13|14","e":{"fu":"F1","be":true,"bu":"B1"}}]
```

**Comment by tuanchauict** [comment]
Fixed on 1.2.1
<img width="308" alt="image" src="https://github.com/tuanchauict/MonoSketch/assets/955306/5045613b-bb79-4dc8-866c-0a27a4855f3e">
